src/main/java/manufacturing/web/filter/AuthenticationFilter.java

```diff
@@ -0,0 +1,48 @@
+package manufacturing.web.filter;
+
+import java.io.IOException;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import manufacturing.lib.Injector;
+import manufacturing.service.DriverService;
+
+public class AuthenticationFilter implements Filter {
+    private static final String DRIVER_ID = "driver_id";
+    private static final Injector injector = Injector.getInstance("manufacturing");
+    private final DriverService driverService =
+            (DriverService) injector.getInstance(DriverService.class);
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse,
+                         FilterChain filterChain) throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) servletRequest;
+        HttpServletResponse resp = (HttpServletResponse) servletResponse;
+        String url = req.getServletPath();
+        if (url.equals("/login") || url.equals("/drivers/create")) {
+            filterChain.doFilter(req, resp);
+            return;
+        }
+        Long driverId = (Long) req.getSession().getAttribute(DRIVER_ID);
+        if (driverId == null || driverService.get(driverId) == null) {
```

Comment:
https://mate-academy.github.io/jv-program-fulltime/02_jdbc/checklist/09_filters_checklist.html
- 5 paragraph

Reply:
sorry. didn't understand this. done